### ISSUE TITLE
There is a problem with merging models

### ISSUE THREAD

**Issue by DubovitskyDenis** [colmap/colmap#1917]
I split one set of 35 photos into two that have a certain number of photos in common in several ways:
1) 3 common images (merge failed)
2) 6 common images (merge failed)
3) 9 common images (merge failed)
4) 12 common images (merge failed)
5) 15 common images also merge failed, but there was an error:
What could be causing this? Any advice on what part the submodels should have in common?

![image](https://github.com/colmap/colmap/assets/20275370/2fc4d017-b4b6-4893-b19f-d7e0cafb8203)

**Comment by DubovitskyDenis** [comment]
With 18 common images merged succeeded, but 18 common image of 35 unique still big overlapping. Maybe there is any hints for good splitting to submodels?

**Comment by ferreram** [comment]
The model merger uses common image ids and 3D points ids between submodels to perform the alignment.  Hence, if you have processed both reconstruction separately (i.e. they each have a different database), common images between both reconstructions won't have the same image ids and same goes for the 3D points.

**Comment by DubovitskyDenis** [comment]
No, I use only one database (of course I read this advice https://github.com/colmap/colmap/issues/1703#issuecomment-1331828759), but still 18 of 35 images using as common is a big overlapping.

**Comment by ferreram** [comment]
DId you try to increase the max_reproj_error?  The merge is only accepted if enough common 3D points (30% I think) gets projected through the similarity transformation in images from submodel 1 and submodel 2 with a lower reprojection error that this maximum threshold.

**Comment by DubovitskyDenis** [comment]
With increasing the max_reproj_error it works for smaller common image sets it can work. But the result of merging is not perfect. Anyway, thank you for advice!

**Comment by ahojnnes** [comment]
Closing: model merging question answered; user confirmed max_reproj_error workaround resolved it.
